File: Code Examples/KeyAPI/src/main/java/com/example/security/token/ApiKeyTokenGenerator.java
package com.example.security.token;


import jakarta.inject.Singleton;
import org.jasypt.encryption.pbe.StandardPBEStringEncryptor;

import java.time.Instant;

@Singleton
public class ApiKeyTokenGenerator {

    public String generateKey(String name,String password,  Instant time)
    {

        StandardPBEStringEncryptor encryptor = new StandardPBEStringEncryptor();

        encryptor.setPassword(password);
        encryptor.setAlgorithm("PBEWithMD5AndTripleDES");
        return  encryptor.encrypt(new StringBuilder()
                .append(name)
                .append( time.toString()).toString());

    }
}

File: MicroCli-deprecated/src/main/resources/micronaut/security/api key/kotlin/JDBC/token/ApiKeyTokenGenerator.kt
package ${securityPackage}.token

import jakarta.inject.Singleton
import org.jasypt.encryption.pbe.StandardPBEStringEncryptor
import java.time.Instant

@Singleton
class ApiKeyTokenGenerator {
    fun generateKey(name: String?, password: String?, time: Instant): String {
        val encryptor = StandardPBEStringEncryptor()
        encryptor.setPassword(password)
        encryptor.setAlgorithm("PBEWithMD5AndTripleDES")
        return encryptor.encrypt(
            StringBuilder()
                .append(name)
                .append(time.toString()).toString()
        )
    }
}